### PR TITLE
V2 dev -> master

### PR DESCRIPTION
**Pull request by jerrysummer**
https://github.com/SynapseFI/SynapseNode/pull/115
https://github.com/SynapseFI/SynapseNode/pull/108
https://github.com/SynapseFI/SynapseNode/pull/113

### CODE REVIEW

**Review by japamatSynapse — APPROVED** [review]
Lots of changes - but should make the SDK more developer friendly. Moved to typescript, consistent comments (with links to docs), typing.

From a code perspective, a few redundant files were removed.

Approved

File: src/lib/Client.ts [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+  getNodeTypes() {
+    const { host } = this;
+    return axios.patch(`${host}/nodes/types`);
+  }
+
+  /**
+   * @description {@link [Fetches allowed user document types](https://docs.synapsefi.com/api-references/users/allowed-document-types)}
+   * @returns list of allowed user document types
+   */
+  getUserDocumentTypes() {
```
Comment (sheilf-synapse) [inline src/lib/Client.ts:412]: is `.patch` expected in these client calls? The document writes that it's a `GET` call

eg:
https://docs.synapsefi.com/api-references/nodes/allowed-node-types

Comment (japamatSynapse) [reply src/lib/Client.ts:412]: good catch - fixed

**Review by japamatSynapse — APPROVED** [review]
approved

**Review by sheilf-synapse — APPROVED** [review]

**Pull request merged**